tanner/tests/test_api.py

```diff
@@ -0,0 +1,224 @@
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_content, self.expected_content)
+
+    def test_return_snare_info(self):
+        query_res = ['{"sess1": {"end_time": 2.00, "start_time": 0.00 }, "sess2": {"attack_types": ["rfi"]}}']
+
+        self.redis_client.zrevrangebyscore = AsyncMock(return_value=query_res)
+        self.handler = Api(self.redis_client)
+        self.expected_content = [{'sess1': {'end_time': 2.0, 'start_time': 0.0}, 'sess2': {'attack_types': ['rfi']}}]
+
+        async def test():
+            self.returned_content = await self.handler.return_snare_info(self.uuid)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_return_snare_info_error(self):
+        self.redis_client.zrevrangebyscore = AsyncMock(side_effect=aioredis.ProtocolError)
+        self.handler = Api(self.redis_client)
+
+        async def test():
+            self.returned_content = await self.handler.return_snare_info(self.uuid)
+
+        with self.assertLogs(level='ERROR') as log:
+            self.loop.run_until_complete(test())
+            self.assertIn('Can not connect to redis', log.output[0])
+
+    def test_return_session_info(self):
+        self.handler = Api(self.redis_client)
+
+        sessions = {
+            "sess1": {
+                "sess_uuid": "c546114f97f548f982756495f963e280"
+            },
+            "sess2": {
+                "sess_uuid": "f432785f97f548f98289054f963e972"
+            }
+        }
+        self.handler.return_snare_info = AsyncMock(return_value=sessions)
+        self.expected_content = 'sess1'
+
+        async def test():
+            self.returned_content = await self.handler.return_session_info(self.uuid, self.snare_uuid)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_content, self.expected_content)
+
+    def test_return_sessions(self):
+        self.handler = Api(self.redis_client)
+        self.handler.return_snares = AsyncMock(return_value=["8fa6aa98-4283-4085-bfb9-a1cd3a9e56e4"])
+
+        sessions = {
+            "sess1": {
+                'user_agent': "Mozilla/5.0",
+                'peer_ip': "10.0.0.3"
+            },
+            "sess2": {
+                'attack_types': ["lfi", "xss"],
+                'possible_owners': ["crawler"],
+                'start_time': 148580,
+                'end_time': 148588,
+                'snare_uuid': [self.snare_uuid]
+            }
+        }
+
+        self.handler.return_snare_info = AsyncMock(return_value=sessions)
+        self.filters = {
+            'user_agent': "Mozilla",
+            'peer_ip': "10.0.0.1",
+            'attack_types': "xss",
+            'possible_owners': "crawler",
+            'start_time': 148575,
+            'end_time': 148590,
+            'snare_uuid': self.snare_uuid
+        }
+
+        def mock_apply_filter(filter_name, filter_value, session):
```

Comment:
I guess you can return false in some cases just on the basis of the session's name, like
```
if name==sess2 or sess 3:
    return False
else:
    return True
```

Reply:
Also check, `apply_filter` is called with all the above filters.

Reply:
I tried checking this in a loop but after running the method in `run_until_complete` it only returns the last filter of the last session not the current filters, session of the loop.

Reply:
[This](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls) might help.